### PR TITLE
Feature/add admin api

### PR DESCRIPTION
**Pull request by jm-clius**
Partial implementation of #301.

#### What is added?
It adds a method to return a list of peers registered on Waku node as per https://github.com/vacp2p/specs/issues/259. For now it simply returns whichever is available of `FilterPeer`, `HistoryPeer` and `SwapPeer` in a single list.

#### What is outstanding?
Returning peers connected for `WakuRelay`. Since the underlying switch does not expose its connections, this requires some peer storing/tracking as in https://github.com/status-im/nim-waku/issues/265

### CODE REVIEW

File: waku/v2/node/jsonrpc/admin_api.nim [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # @TODO since the switch does not expose its connections, retrieving the connected peers requires a peer store/peer management
+
+    var wPeers: seq[WakuPeer] = @[]
+
+    if not node.wakuSwap.isNil:
+      # Map WakuSwap peers to WakuPeers and add to return list
+      wPeers.insert(node.wakuSwap.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
+                                                       protocol: WakuSwapCodec,
+                                                       connected: node.switch.isConnected(it.peerInfo))),
+                    wPeers.len) # Append to the end of the sequence
```
Comment (staheri14) [inline waku/v2/node/jsonrpc/admin_api.nim:36]:
```suggestion
      result.insert(node.wakuSwap.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
                                                       protocol: WakuSwapCodec,
                                                       connected: node.switch.isConnected(it.peerInfo))),
                    result.len) # Append to the end of the sequence
```

File: waku/v2/node/jsonrpc/admin_api.nim [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    
+    if not node.wakuFilter.isNil:
+      # Map WakuFilter peers to WakuPeers and add to return list
+      wPeers.insert(node.wakuFilter.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
+                                                         protocol: WakuFilterCodec,
+                                                         connected: node.switch.isConnected(it.peerInfo))),
+                    wPeers.len) # Append to the end of the sequence
+
```
Comment (staheri14) [inline waku/v2/node/jsonrpc/admin_api.nim:44]:
```suggestion
      result.insert(node.wakuFilter.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
                                                         protocol: WakuFilterCodec,
                                                         connected: node.switch.isConnected(it.peerInfo))),
                    result.len) # Append to the end of the sequence

```

File: waku/v2/node/jsonrpc/admin_api.nim [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if not node.wakuStore.isNil:
+      # Map WakuStore peers to WakuPeers and add to return list
+      wPeers.insert(node.wakuStore.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
+                                                        protocol: WakuStoreCodec,
+                                                        connected: node.switch.isConnected(it.peerInfo))),
+                    wPeers.len) # Append to the end of the sequence
```
Comment (staheri14) [inline waku/v2/node/jsonrpc/admin_api.nim:50]:
```suggestion
      result.insert(node.wakuStore.peers.mapIt(WakuPeer(multiaddr: constructMultiaddrStr(it.peerInfo),
                                                        protocol: WakuStoreCodec,
                                                        connected: node.switch.isConnected(it.peerInfo))),
                    result.len) # Append to the end of the sequence
```

File: waku/v2/node/jsonrpc/admin_api.nim [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+    # @TODO filter output on protocol/connected-status
+    return wPeers
```
Comment (staheri14) [inline waku/v2/node/jsonrpc/admin_api.nim:54]:
```suggestion

```

File: waku/v2/node/jsonrpc/admin_api.nim [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  ## Admin API version 1 definitions
+
+  rpcsrv.rpc("get_waku_v2_admin_v1_peers") do() -> seq[WakuPeer]:
+    ## Returns history for a list of content topics with optional paging
+    debug "get_waku_v2_admin_v1_peers"
+
+    # Create a single list of peers from mounted protocols.
+    # @TODO since the switch does not expose its connections, retrieving the connected peers requires a peer store/peer management
+
+    var wPeers: seq[WakuPeer] = @[]
```
Comment (staheri14) [inline waku/v2/node/jsonrpc/admin_api.nim:29]:
```suggestion
```

**Review by staheri14 — CHANGES_REQUESTED** [review]

**Comment by staheri14** [comment]
Looks good to me!

**Review by staheri14 — COMMENTED** [review]
Looks good to me, just added a few suggestions (also changed it from requested changes)

**Review by staheri14 — COMMENTED** [review]
Looks good to me, just added a few suggestions (also changed it from requested changes)

**Comment by jm-clius** [comment]
Thanks, @staheri14! We are trying to avoid using `result` as it suppresses useful compiler warnings. It's still used commonly elsewhere, so must be refactored (see https://github.com/status-im/nim-waku/issues/287). The alternative for me here was to return all the selection, filtering and concat logic bundled in a single expression, but I felt it became unreadable. The compromise was for me to use this local variable and be explicit about how it's populated.

**Comment by staheri14** [comment]
> Thanks, @staheri14! We are trying to avoid using `result` as it suppresses useful compiler warnings. It's still used commonly elsewhere, so must be refactored (see #287). The alternative for me here was to return all the selection, filtering and concat logic bundled in a single expression, but I felt it became unreadable. The compromise was for me to use this local variable and be explicit about how it's populated.

I see, right! it is good to be refactored  to reflect the new style

**Review by staheri14 — APPROVED** [review]

**Review by oskarth — APPROVED** [review]
LGTM!

**Comment by oskarth** [comment]
Oh, please update CHANGELOG as well

**Comment by jm-clius** [comment]
> Oh, please update CHANGELOG as well (one-liner under `Next` section so releases are easier)

Done in d49736b

**Pull request merged**